client/src/components/Donation/StripeCardForm.js

```diff
@@ -1,88 +1,159 @@
-import { CardNumberElement, CardExpiryElement } from 'react-stripe-elements';
+import isEmail from 'validator/lib/isEmail';
+
 import {
   Row,
   Col,
   ControlLabel,
   FormGroup,
-  Image
+  Image,
+  Button,
+  Form,
+  FormControl,
+  Alert
 } from '@freecodecamp/react-bootstrap';
 import { withTranslation } from 'react-i18next';
 
+const initialPaymentInfoValidityState = {
+  cardNumber: {
+    complete: false,
+    error: null
+  },
+  cardExpiry: {
+    complete: false,
+    error: null
+  }
+};
+
 const propTypes = {
-  getValidationState: PropTypes.func.isRequired,
+  getDonationButtonLabel: PropTypes.func.isRequired,
+  onDonationStateChange: PropTypes.func,
+  postStripeDonation: PropTypes.func,
   t: PropTypes.func.isRequired,
-  theme: PropTypes.string
+  theme: PropTypes.string,
+  userEmail: PropTypes.string
 };
 
-const style = {
-  base: {
-    fontSize: '18px'
-  }
-};
+const StripeCardForm = ({
+  getDonationButtonLabel,
+  theme,
+  t,
+  onDonationStateChange,
+  postStripeDonation,
+  userEmail
+}) => {
+  const [isSubmissionValid, setSubmitionValidity] = useState(true);
+  const [email, setEmail] = useState(userEmail);
+  const [isEmailValid, setEmailValidity] = useState(true);
+  const [paymentInfoValidation, validitySetter] = useState(
+    initialPaymentInfoValidityState
+  );
 
-class StripeCardForm extends Component {
-  constructor(...props) {
-    super(...props);
-
-    this.state = {
-      validation: {
-        cardNumber: {
-          complete: false,
-          error: null
-        },
-        cardExpiry: {
-          complete: false,
-          error: null
-        }
-      }
-    };
+  const stripe = useStripe();
+  const elements = useElements();
 
-    this.handleInputChange = this.handleInputChange.bind(this);
-    this.isValidInput = this.isValidInput.bind(this);
+  function handleInputChange(event) {
+    const { elementType, error, complete } = event;
+    validitySetter({
+      ...paymentInfoValidation,
+      [elementType]: {
+        error,
+        complete
+      }
+    });
   }
 
-  componentDidMount() {
-    this.props.getValidationState(this.isValidInput());
+  function isPaymentInfoValid() {
+    return Object.keys(paymentInfoValidation)
+      .map(key => paymentInfoValidation[key])
+      .every(({ complete, error }) => complete && !error);
   }
 
-  handleInputChange(event) {
-    const { elementType, error, complete } = event;
-    return this.setState(
-      state => ({
-        ...state,
-        validation: {
-          ...state.validation,
-          [elementType]: {
-            error,
-            complete
-          }
-        }
-      }),
-      () => this.props.getValidationState(this.isValidInput())
+  const options = {
+    style: {
+      base: {
+        fontSize: '18px',
+        color: `${theme === 'night' ? '#fff' : '#0a0a23'}`
+      }
+    }
+  };
+
+  const handleSubmit = async event => {
+    event.preventDefault();
+
+    if ((!isEmailValid, !isPaymentInfoValid()))
```

Comment:
```suggestion
    if (!isEmailValid || !isPaymentInfoValid())
```
This was quite interesting - I didn't know there was a [comma operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comma_Operator)

However, I think we just wanted it to be invalid if either of the email or payment info were invalid, right?

Reply:
right. thanks for looking into this.